apps/api/src/deployment/services/deployment/deployment.service.ts

```diff
@@ -0,0 +1,83 @@
+import { MsgCreateDeployment } from "@akashnetwork/akash-api/v1beta3";
+import { SDL } from "@akashnetwork/akashjs/build/sdl";
+import { getAkashTypeRegistry } from "@akashnetwork/akashjs/build/stargate";
+import { DirectSecp256k1HdWallet, Registry } from "@cosmjs/proto-signing";
+import { SigningStargateClient } from "@cosmjs/stargate";
+import axios from "axios";
+import { singleton } from "tsyringe";
+
+import { BillingConfigService } from "@src/billing/services/billing-config/billing-config.service";
+import { SingleDeploymentResponse } from "@src/deployment/http-schemas/deployment.schema";
+import { MessageTransmitterService } from "@src/deployment/services/message-transmitter/message-transmitter.service";
+import { RestAkasheploymentInfoResponse, RestAkashLeaseListResponse } from "@src/types/rest";
+import { apiNodeUrl, betaTypeVersion, betaTypeVersionMarket } from "@src/utils/constants";
+import { env } from "@src/utils/env";
+
+@singleton()
+export class DeploymentService {
+  constructor(
+    private readonly config: BillingConfigService,
+    private readonly messageTransmitterService: MessageTransmitterService,
+  ) { }
+
+  public async findByOwnerAndDseq(owner: string, dseq: string): Promise<SingleDeploymentResponse['data']> {
+    const response = await axios.get<RestAkasheploymentInfoResponse>(`${apiNodeUrl}/akash/deployment/${betaTypeVersion}/deployments/info`, {
+      params: {
+        "id.owner": owner,
+        "id.dseq": dseq,
+      }
+    });
+
+    if ('code' in response.data) {
+      return {
```

Comment:
why do we need to resolve with this response? I assume code may be different? controller should throw appropriate http error. E.g. if deployment not found I assume this return value should be `undefined` and controller should throw 404. If it's smth else it should be 500 perhaps.